### ISSUE TITLE
Error: connection refused: localtunnel.me:44972 (check your firewall settings)

### ISSUE THREAD

**Issue by ElijahLynn** [localtunnel/localtunnel#258]
Rubber ducking this here. Latest version from npm, 1.9.0. 

`lt --port 80 --subdomain elijah --local-host "local.example.com"`
your url is: https://elijah.localtunnel.me
/home/elijah/.config/fnm/lib/node_modules/localtunnel/bin/client:65
        throw err;
        ^

Error: connection refused: localtunnel.me:44972 (check your firewall settings)
    at Socket.<anonymous> (/home/elijah/.config/fnm/lib/node_modules/localtunnel/lib/TunnelCluster.js:47:32)
    at Socket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)

**Comment by ElijahLynn** [comment]
Hrm, tried 3 more times, 2nd time same error, 3rd time no error but exited with no error, 4th time worked. Then 5th time it gave me a generated subdomain, not the `--subdomain elijah` I requested. Maybe because the server was still holding on to the old subdomain.

**Comment by olufemioyedepo** [comment]
facing the same issues...

**Comment by ElijahLynn** [comment]
It definitely appears to be on the serverside. Probably best to setup a server instance for my own use as the public server may not be robust enough to handle all the public usage. And then I could look at the logs when/if this stuff happens too.

**Comment by BananaAcid** [comment]
Happens to me after a few requests - initiating always works fine.

**Comment by BerndErnst** [comment]
So @defunctzombie is simply going cheap on us

**Comment by defunctzombie** [comment]
@BerndErnst Please see https://github.com/localtunnel/localtunnel/issues/261#issuecomment-407550521

**Comment by ElijahLynn** [comment]
@BerndErnst You may not have meant your comment to sound this way but it sounds unappreciative to me. This is a free service, the code is open source and we can spin up our own servers if we like and @defunctzombie doesn't really owe anyone a high availability server or bandwidth. 

That being said, I think the action item for this particular issue is to make the error message a bit better. I propose the message to simply be changed from > to:

> Error: connection refused: localtunnel.me:44972 (check your firewall settings) 

> Error: connection refused: localtunnel.me:44972 (check your firewall settings or server may be under high load, try again in a few minutes) 

And maybe distinguish between an external connection refusal vs a firewall issue for isolated error messages.

**Comment by pierrebonbon** [comment]
Getting the same error. Here's an example log:

    your url is: https://afraid-baboon-40.localtunnel.me
    /usr/lib64/node_modules/localtunnel/bin/client:65
        throw err;
        ^

    Error: connection refused: localtunnel.me:37161 (check your firewall settings)
    at Socket.<anonymous> (/usr/lib64/node_modules/localtunnel/lib/TunnelCluster.js:47:32)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)

**Comment by ElijahLynn** [comment]
@pierrebonbon If you wait a minute and try again, repeat. You should eventually get a connection established.

**Comment by piotrkochan** [comment]
but every time new address is generated... that's making things complicated.

**Comment by ElijahLynn** [comment]
If you pass a name in then it will use the same one `lt --subdomain yellowtruck`.

**Comment by juloko** [comment]
This issue still present.

**Comment by mariale425** [comment]
For me too. I have that issue

**Comment by ElijahLynn** [comment]
https://github.com/localtunnel/localtunnel/issues/261#issuecomment-407550521

> I am glad to hear this project is useful! However, I must also note that this project is a low priority side project for me. I come back to it from time to time (this is why some issues get closed, etc) but generally I would not expect a fast response from me here.
> 
> For those that do find the project useful but suffer from the flaky public server; the server is open source https://github.com/localtunnel/server and I recommend that if you have serious needs you set it up yourself. The client has a command line argument to use other servers.
> 
> I will continue to run the public server and make small tweaks here or there if I desire - however I would not consider this project _actively maintained_.

tldr; The issue is that the server is overloaded, it is a side project, what community needs is someone who wants to step up and maintain a new server, and pay for it.

**Comment by kevin-06-huang** [comment]
is there a way to change the host programmatically in node.js? for example, I have these lines 
"var tunnel = localtunnel(port, { subdomain: 'embers'}, function(err, tunnel) {
  ...
});"

where would I put the option to change the host to a self-hosted server that I'm running, instead of the localtunnel.me server?

**Comment by ElijahLynn** [comment]
> is there a way to change the host programmatically in node.js? for example, I have these lines
> "var tunnel = localtunnel(port, { subdomain: 'embers'}, function(err, tunnel) {
> ...
> });"
> 
> where would I put the option to change the host to a self-hosted server that I'm running, instead of the localtunnel.me server?

Looks like it might be here:

https://github.com/localtunnel/localtunnel/blob/a4bf6d16e4e7be292bb90bd100644cabe46b7543/lib/Tunnel.js#L17

**Comment by pushpak1300** [comment]
I think it is due to server is facing some issues.

**Comment by Mridul-parakh** [comment]
At the terminal, run npm install --save-dev forever 

In your root project directory, create a new file called sendgrid_webhook.js .  Add the following to that file:

var localtunnel = require('localtunnel');
localtunnel(5000, { subdomain: <YOUR_SUBDOMAIN> }, function(err, tunnel) {
  console.log('LT running')
});
Replace <YOUR_SUBDOMAIN> with the subdomain name you picked, so it should look something like 

{ subdomain: 'asdfasdf' }
In the package.json file, replace the "webhook" script with the following:

"webhook": "forever sendgrid_webhook.js"

**Comment by Mridul-parakh** [comment]
it will help to restart again

**Comment by LWFlouisa** [comment]
I still get the issue:

/usr/local/lib/node_modules/localtunnel/bin/client:65
        throw err;
        ^

Error: connection refused: localtunnel.me:39205 (check your firewall settings)
    at Socket.<anonymous> (/usr/local/lib/node_modules/localtunnel/lib/TunnelCluster.js:47:32)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)

It initially sets up just fine, but then crashes.

**Comment by pablolopesk8** [comment]
I've the same error. And really, the only way is restart the service, unfortunately!

**Comment by vallme2003** [comment]
Got the same error. Restarting the service works.

/home/linuxbrew/.linuxbrew/lib/node_modules/localtunnel/bin/client:65
        throw err;
        ^

Error: connection refused: localtunnel.me:36317 (check your firewall settings)
    at Socket.<anonymous> (/home/linuxbrew/.linuxbrew/lib/node_modules/localtunnel/lib/TunnelCluster.js:47:32)
    at Socket.emit (events.js:203:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at processTicksAndRejections (internal/process/task_queues.js:77:11)

**Comment by xhalo32** [comment]
Here's a workaround with bash:
```bash
until npx lt <options>; do done
```

**Comment by vicke4** [comment]
@xhalo32 could you tell us how it works?

**Comment by shamoons** [comment]
@vicke4 just restarts `lt` until it works

**Comment by vicke4** [comment]
@shamoons I can understand that. I wanted to know how restarting fixes this issue.

**Comment by alfonmga** [comment]
localtunnel.sh:
```sh
#!/bin/bash 
until lt --port 3000 --subdomain amga-monorepo-starter-api
do
  echo "Try again"
done
```

Example of how I use it in a project: https://github.com/AMGAVentures/monorepo-starter/blob/46d393a02033f325862c8e95451059d7efd3f34d/packages/api/package.json#L6

**Comment by Mykola-Veriga** [comment]
I just used `sudo`  and it is working

**Comment by defunctzombie** [comment]
The localtunnel server experiences times of high load and instability because it is a free public service. Often retrying (as suggested in the comments) helps.

If you need a reliable test server you can setup your own localtunnel server backed (see the server repo).

**Comment by jelling** [comment]
I reverted to localhost@1.8.3 and it started working again. 2.0 did not work for me.

**Comment by zcmgyu** [comment]
I've created my own server but still get this issue.

**Comment by mchlkucera** [comment]
This issue is still present. Even after multiple restarts. doesn't matter if I go with --subdomain or not

**Comment by codecolony** [comment]
C:\Users\<user>\AppData\Roaming\npm\node_modules\localtunnel\bin\lt.js:81
    throw err;
    ^

Error: connection refused: localtunnel.me:42370 (check your firewall settings)
    at Socket.<anonymous> (C:\Users\<user>\AppData\Roaming\npm\node_modules\localtunnel\lib\TunnelCluster.js:52:11)
    at Socket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
    
I get this error. I don't think it is due to server overload.

**Comment by mustafayuce33** [comment]
Same issue here

**Comment by JexPY** [comment]
Same issue on mac.

**Comment by hosjiu1702** [comment]
> I just used `sudo` and it is working

 I don't know why this works well? 

any explanation for this behavior, thanks.

**Comment by aludayalu** [comment]
Its not working at my side too :((
even sudo doesnt help
lt --port 80 --subdomain xxxxxx
it outputs:
your url is: https://xxxxxx.loca.lt
/usr/lib/node_modules/localtunnel/bin/lt.js:81
    throw err;
    ^

Error: connection refused: localtunnel.me:44007 (check your firewall settings)
    at Socket.<anonymous> (/usr/lib/node_modules/localtunnel/lib/TunnelCluster.js:52:11)
    at Socket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:164:8)
    at emitErrorCloseNT (node:internal/streams/destroy:129:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)


Someone please help

**Comment by frontendreact** [comment]
I tried to clear caches, and then generate a new tl link, it worked for me.

**Comment by SX-9** [comment]
So the problem is that your antivirus firewall is blocking the connection, try temporary disabling it!

**Comment by SHARKZTECH** [comment]
> C:\Users\\AppData\Roaming\npm\node_modules\localtunnel\bin\lt.js:81 throw err; ^
> 
> Error: connection refused: localtunnel.me:42370 (check your firewall settings) at Socket. (C:\Users\\AppData\Roaming\npm\node_modules\localtunnel\lib\TunnelCluster.js:52:11) at Socket.emit (events.js:315:20) at emitErrorNT (internal/streams/destroy.js:106:8) at emitErrorCloseNT (internal/streams/destroy.js:74:3) at processTicksAndRejections (internal/process/task_queues.js:80:21)
> 
> I get this error. I don't think it is due to server overload.
try  `sudo lt --port <PORT>`  if you are experiencing the same error.

**Comment by Aarush-Acharya** [comment]
hello i am facng the error normally also 

❯ lt --port 85
your url is: https://kind-wings-change.loca.lt
/usr/local/lib/node_modules/localtunnel/bin/lt.js:81
    throw err;
    ^

Error: connection refused: localtunnel.me:43471 (check your firewall settings)
    at Socket.<anonymous> (/usr/local/lib/node_modules/localtunnel/lib/TunnelCluster.js:52:11)
    at Socket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

Node.js v18.13.0

**Comment by UmarNoor** [comment]
Guys - this is an unreliable, not-maintained piece of software as author himself has mentioned. I appreciate the work though but if you are facing this firewall issues, try using **ssh tunnel**... it works like a charm and is absolutely free of cost....

**Comment by davidsonsns** [comment]
> localtunnel.sh:
> 
> ```shell
> #!/bin/bash 
> until lt --port 3000 --subdomain amga-monorepo-starter-api
> do
>   echo "Try again"
> done
> ```
> 
> Example of how I use it in a project: https://github.com/AMGAVentures/monorepo-starter/blob/46d393a02033f325862c8e95451059d7efd3f34d/packages/api/package.json#L6

@alfonmga thanks for sharing your solution. it worked beautifully. I just made some small changes so I can keep up with the attempts

```shell
#!/bin/bash 
echo "Starting localtunnel"

attempts=0
until lt --port 4000 --subdomain YOUR_SUBDOMAIN --print-requests
do
  attempts=$((attempts+1))
  echo "Attempt $attempts failed. Try again."
done

echo "Success after $attempts attempts."
```

**Comment by avii09** [comment]
it worked at the beginning for me and then all of a sudden i get this error?

**Comment by devdrops** [comment]
Same issue. I understand the issue, and the state of the project (I'm not complaining at all, just trying a solution for my case). Trying with the command below:

```sh
watch -n 10 -c lt --port 9080 --subdomain angrypotato
```

I'll update with the results :wink: 

**Update**: it worked at first try after restarting the computer :upside_down_face:

**Comment by cheesecode** [comment]
> it worked at the beginning for me and then all of a sudden i get this error?

Same, it works for a while but after a couple of hours it breaks.

**Comment by Hube2000** [comment]
i believed it's link to the firwall on the network

**Comment by sachinshan97** [comment]
The below flag works for me. But it goes to muted state when opting for a subdomain. 

- Just Worked: 
```
 lt --local-host 0.0.0.0 --port 8090
your url is: https://xyz-help.loca.lt
```
- Response:
```
 curl https://xyz-help.loca.lt
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
<meta h
```

- Subdomain: [Can't get a public URL]
```
 lt --local-host 0.0.0.0 --port 8090 -s somename

```
